lib/FileModel.php

```diff
@@ -3,117 +3,47 @@
 namespace OCA\Analysis_app;
 
 use OCP\Files\Folder;
-use OCP\Files\Node;
 
 class FileModel {
-
     /**
-     * @var array $fileMimeTypesInfo
-     * @var array $sortedFileSizes
+     * @var array[][] $fileEverything
      */
-    private $fileMimeTypesInfo;
-    private $fileSortingAsSize;
+    private $fileEverything;
+
     /**
      * FileModel constructor.
      */
     public function __construct() {
-      /*  $this->determineFileMimeTypesInfo(\OC::$server->getUserFolder());
-        $this->determineBiggestFiles(\OC::$server->getUserFolder());*/
-        $this->returnEverything(\OC::$server->getUserFolder());
-    }
-
-
-    /**
-     * @param Folder $currentDirectory
-     */
-    public function determineFileMimeTypesInfo($currentDirectory) {
-        foreach ($currentDirectory->search('') as $item){
-            if ($item->getMimetype() == 'http/unix-directory') {
-                $this->determineFileMimeTypesInfo($item);
-            } else {
-                if(!isset($this->fileMimeTypesInfo[$item->getMimetype()])) {
-                    $this->fileMimeTypesInfo[$item->getMimetype()] = $item->getSize();
-                } else {
-                    $this->fileMimeTypesInfo[$item->getMimetype()] += $item->getSize();
-                }
-            }
-        }
+        $this->determineEverything(\OC::$server->getUserFolder());
     }
-
-
     /**
      * @param Folder $currentDirectory
      */
-    public function determineBiggestFiles($currentDirectory) {
+    public function determineEverything($currentDirectory){
         foreach ($currentDirectory->search('') as $item) {
             if ($item->getMimetype() == 'http/unix-directory') {
-                $this->determineBiggestFiles($item);
+                $this->determineEverything($item);
             } else {
-                $this->fileSortingAsSize[$item->getName()] = $item->getSize();
-            }
-        }
-    }
 
-    /**
-     * @param Folder $currentDirectory
-     */
-    public function determineEveryting($currentDirectory){
-        foreach ($currentDirectory->search('') as $item) {
-            if ($item->getMimetype() == 'http/unix-directory') {
-                $this->determineEveryting($item);
-            } else {
-                $this->fileSortingAsSize[$item->getName()] = $item->getSize();
-                if(!isset($this->fileMimeTypesInfo[$item->getMimetype()])) {
-                    $this->fileMimeTypesInfo[$item->getMimetype()] = $item->getSize();
+                if(count($this->fileEverything['fileList'])<10
+                    || $item->getSize()>array_values($this->fileEverything['fileList'][9])
+                ) {
+
+                    $this->fileEverything['fileList'][$item->getPath()]= $item->getSize();

```

Comment:
Also when I tried to assign file as $item, not $item->getSize() , it didn't work, all application crashed, and relatively sort algorithm couldn't work. So basically I wanted to see something that works at the moment, so I changed it this way for short notice.